BedJet - Can't connect to my Bedjet V3

### The problem

Not connected, will not send status.

![image](https://user-images.githubusercontent.com/166283/202597570-a527b66f-11f1-413d-8b12-4cc6db5b9e4e.png)


### Which version of ESPHome has the issue?

2022.11.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.11.2

### What platform are you using?

ESP32

### Board

Node-MCU

### Component causing the issue

Bedjet

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
This is all I have at the moment on the code.

I know I don't have the climate and fan, jsut trying to debug, puttin gless info possible.

When I can for BLE, I see my BEDJET v3, and the mac address is good. Check over 10 times.
```


### Additional information

substitutions:
  name: esp32-ble-proxy-and-bedjet
packages:
  esphome.bluetooth-proxy: github://esphome/bluetooth-proxies/esp32-generic.yaml@main
esphome:
  name: ${name}
  name_add_mac_suffix: false


wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

ble_client:
  - mac_address: 94:B5:55:30:E8:1E
    id: BEDJET_V3

bedjet:
  - id: bedjet_1
    ble_client_id: BEDJET_V3

> @nochlab1 that is helpful - I have a few more log statements to verify my suspicions, if you wouldn't mind rebuilding and redeploying from the same branch, and report back what logs you get.
> 
> I suspect that this is a race condition between the component setup order: BLEClient is setup using AFTER_BLUETOOTH, while BedJet is setup with AFTER_WIFI. BLEClient gets setup first, and then BedJet later. Normally that's probably not a problem, but if the BedJet is actively broadcasting notification packets (while there's no client listening to those packets), then as soon as BLEClient is initialized, it will start trying to dispatch these gatt events, including the notify packet. But if BedJet hasn't been initialized yet, it will be trying to handle a packet on an incomplete component, so the memory address will be uninitialized (null / 0), and trying to access the status packet buffer on that, results in a memory address of `0xc` which is invalid, hence the crash you're seeing.

it didnt build 
INFO Reading configuration /config/esphome/bedjet.yaml...
INFO Updating https://github.com/jhansche/esphome.git@test/3807
INFO Generating C++ source...
INFO Compiling app...
Processing bedjet (board: esp32dev; framework: arduino; platform: platformio/espressif32 @ 5.2.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- AsyncTCP-esphome @ 1.2.2
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- AsyncTCP-esphome @ 1.2.2
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.4
|   |-- libsodium @ 1.10018.1
Compiling /data/bedjet/.pioenvs/bedjet/src/esphome/components/bedjet/bedjet_codec.cpp.o
In file included from src/esphome/components/bedjet/bedjet_codec.h:4,
                 from src/esphome/components/bedjet/bedjet_codec.cpp:1:
src/esphome/components/bedjet/bedjet_codec.cpp: In member function 'bool esphome::bedjet::BedjetCodec::decode_notify(const uint8_t*, uint16_t)':
src/esphome/components/bedjet/bedjet_codec.cpp:96:36: error: lvalue required as unary '&' operand
              length, &this->buf_, &this, sizeof(BedjetStatusPacket), sizeof(this->buf_));
                                    ^~~~
src/esphome/core/log.h:92:89: note: in definition of macro 'esph_log_d'
   esp_log_printf_(ESPHOME_LOG_LEVEL_DEBUG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
                                                                                         ^~~~~~~~~~~
src/esphome/components/bedjet/bedjet_codec.cpp:95:5: note: in expansion of macro 'ESP_LOGD'
     ESP_LOGD(TAG, "Writing %d bytes to @%p, on @%p. Clearing 0..%d(%d) first.",
     ^~~~~~~~
*** [/data/bedjet/.pioenvs/bedjet/src/esphome/components/bedjet/bedjet_codec.cpp.o] Error 1
========================== [FAILED] Took 6.12 seconds ==========================


(side point esphome version v2022.9.0 works normally)

--- a/esphome/components/bedjet/bedjet_hub.h
+++ b/esphome/components/bedjet/bedjet_hub.h
@@ -116,7 +116,7 @@ class BedJetHub : public esphome::ble_client::BLEClientNode, public PollingCompo
   void update() override;
   void dump_config() override;
   void setup() override { this->codec_ = make_unique<BedjetCodec>(); }
-  float get_setup_priority() const override { return setup_priority::AFTER_WIFI; }
+  float get_setup_priority() const override { return setup_priority::BLUETOOTH; }
 
   /** @return The BedJet's configured name, or the MAC address if not discovered yet. */
   std::string get_name() {

